### ISSUE TITLE
Sidekiq Enterprise Liveness Health Check Throws Internal Server Error

### ISSUE THREAD

**Issue by wfarr** [sidekiq/sidekiq#6352]
Ruby version: 3.4.0-preview1
Rails version: 7.1.3.2
Sidekiq / Pro / Enterprise version(s): 7.3.0 / 7.3.0 / 7.3.0

Initializer:

```ruby
Sidekiq.configure_server do |config|
  config.redis = {
    url: Rails.application.credentials.dig(:sidekiq, :redis_url) || "redis://localhost:6379/1"
  }

  if Rails.env.test?
    config.logger.level = Logger::WARN
  end
end

Sidekiq.configure_client do |config|
  config.redis = {
    url: Rails.application.credentials.dig(:sidekiq, :redis_url) || "redis://localhost:6379/1"
  }

  if Rails.env.test?
    config.logger.level = Logger::WARN
  end
end
```

sidekiq.yml:

```yaml
---
verbose: false
concurrency: 10
timeout: 25
health_check: "0.0.0.0:7433"

:queues:
  - high
  - default
  - low
```

We've recently moved this Sidekiq into Kubernetes and are attempting to use the liveness probe support in Sidekiq Enterprise.

However, curl yields the following response both from debugging inside the cluster _and_ attempting to reproduce locally (below is an example from my development machine):

```
curl -v http://127.0.0.1:7433/
*   Trying 127.0.0.1:7433...
* Connected to 127.0.0.1 (127.0.0.1) port 7433
> GET / HTTP/1.1
> Host: 127.0.0.1:7433
> User-Agent: curl/8.6.0
> Accept: */*
> 
* HTTP 1.0, assume close after body
< HTTP/1.0 500 Internal Server Error: undefined method 'redis' for main
< Server: Sidekiq::HttpServer (Ruby 3.4.0)
< Connection: close
< Date: Tue, 02 Jul 2024 05:42:15 GMT
< 
* Closing connection
```

I took a quick look at the contents of `lib/sidekiq-ent/liveness.rb` and found the following patch appears to resolve the issue entirely:

```diff
--- lib/sidekiq-ent/liveness.rb.orig	2024-07-02 01:48:19
+++ lib/sidekiq-ent/liveness.rb	2024-07-02 01:42:39
@@ -32,7 +32,7 @@
     def request_handler(req, res)
       return res.status = 404 unless req.path == "/"
 
-      hash = redis { |c| c.hgetall(@key) }
+      hash = Sidekiq.redis { |c| c.hgetall(@key) }
       if hash.size > 0
         res.status = 200
         res.header["Content-Type"] = "application/json"
```

And here's the same local test after this change:

```
curl -v http://127.0.0.1:7433/
*   Trying 127.0.0.1:7433...
* Connected to 127.0.0.1 (127.0.0.1) port 7433
> GET / HTTP/1.1
> Host: 127.0.0.1:7433
> User-Agent: curl/8.6.0
> Accept: */*
> 
* HTTP 1.0, assume close after body
< HTTP/1.0 200 OK
< Server: Sidekiq::HttpServer (Ruby 3.4.0)
< Content-type: application/json
< Content-length: 416
< Connection: close
< Date: Tue, 02 Jul 2024 05:42:51 GMT
< 
{"rtt_us":"605","busy":"0","quiet":"false","rss":"152576","info":{"hostname":"dubbookair.localdomain","started_at":1719898969.050286,"pid":39141,"tag":"<redacted>","concurrency":1,"queues":["high","default","low"],"weights":[{"high":0,"default":0,"low":0}],"labels":[],"identity":"dubbookair.localdomain:39141:c09492882f8f","version":"7.3.0","embedded":false},"beat":"1719898969.060528"}
```

**Comment by mperham** [comment]
Will be fixed in 7.3.1. This was a bug in the config-only processing. If you define the healthcheck in your initializer instead, it will work today:

```ruby
Sidekiq.configure_server do |config|
  config.health_check
end
```

**Comment by wfarr** [comment]
Thanks for the rapid turn-around here! 🙇
